tests_e2e/tests/ext_sequencing/ext_sequencing.py

```diff
@@ -225,7 +225,39 @@ def run(self):
                 # We only expect to catch an exception during deployment if we are forcing one of the extensions to
                 # fail. We know an extension should fail if "failing" is in the case name. Otherwise, report the
                 # failure.
-                deployment_failure_pattern = r"[\s\S]*\"details\": [\s\S]* \"code\": \"(?P<code>.*)\"[\s\S]* \"message\": \"(?P<msg>.*)\"[\s\S]*"
+                #
+                # Example deployment failure:
+                #   (DeploymentFailed) At least one resource deployment operation failed. Please list deployment operations for details. Please see https://aka.ms/arm-deployment-operations for usage details.
+                #   Code: DeploymentFailed
+                #   Message: At least one resource deployment operation failed. Please list deployment operations for details. Please see https://aka.ms/arm-deployment-operations for usage details.
+                #   Exception Details:	(Conflict) {
+                # 	  "status": "Failed",
+                # 	  "error": {
+                # 	    "code": "ResourceDeploymentFailure",
+                # 	    "message": "The resource write operation failed to complete successfully, because it reached terminal provisioning state 'Failed'.",
+                # 	    "details": [
+                # 	      {
+                # 	        "code": "VMExtensionProvisioningError",
+                # 	        "target": "0",
+                # 	        "message": "Multiple VM extensions failed to be provisioned on the VM. The Extensions failed due to the errors: \n[Extension Name: 'AzureMonitorLinuxAgent'\nError Message: VM Extension 'AzureMonitorLinuxAgent' is marked as failed since it depends upon the VM Extension 'CustomScript' which has failed.]\n\n[Extension Name: 'CustomScript'\nError Message: VM has reported a failure when processing extension 'CustomScript' (publisher 'Microsoft.Azure.Extensions' and type 'CustomScript'). Error message: 'Enable failed: failed to execute command: command terminated with exit status=1\n[stdout]\n\n[stderr]\n'. More information on troubleshooting is available at https://aka.ms/VMExtensionCSELinuxTroubleshoot. ]\n"
+                # 	      }
+                # 	    ]
+                # 	  }
+                # 	}
+                deployment_failure_pattern = r"[\s\S]*Exception Details:\t\(Conflict\) {\s*" \
```

Comment:
the regex looks way too specific, possibly breaking if the message is slightly changed. should we parse the json instead and check the properties we are interested in?